### PR TITLE
Tardigrade reference architecture and documentation

### PR DESCRIPTION
**Pull request by confusdcodr**

### CODE REVIEW

File: _site/docs/index.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+└── terragrunt.hcl
+```
+This directory represents the baseline that is replicated to every account being managed
+
+* `account.auto.tfvars` is used to define tenant specific variables
+* `tenant-providers.tf` defines and configures each of the providers to be used by the tenant
+* `terragrunt.hcl` is the terragrunt configuration file
+
+### .../idempotent_element/custom
+This directory is meant to allow you to add in custom elements that a tenant would like included in their account but are outside of the general baseline created for each account.
```
[the remark this answers is not in the document]
Comment (confusdcodr) [reply _site/docs/index.md]: Unfortunately, this was an old build of the site that I created locally and isn't representative of what's actually on the site.

File: docs/_pages/structure.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+│   │       ├── terragrunt.hcl             # terragrunt configuration for the account
+│   │       └── variables.tf               # variable declaration specific to the customization
+│   ├── tenant.tf                          # partition provider and variable declarations
+│   ├── tenant-global.tfvars.yaml          # partition variable values
+│   └── terragrunt.hcl                     # terragrunt configuration
+└── baseline                               # account baseline
+```
+
+## Partition
+In this example, the partition is aws (i.e., `tardigrade/aws`). Generally speaking, most AWS implementations of tardigrade  will only use the `aws` partition. The `aws` parition is the commercial partition that most people are used to. There are other paritions though, such as AWS GovCloud (`aws-us-gov`) and AWS China (`aws-cn`)
```
Comment (lorengordon) [inline docs/_pages/structure.md]:
```suggestion
In this example, the partition is aws (i.e., `tardigrade/aws`). Generally speaking, most AWS implementations of tardigrade  will only use the `aws` partition. The `aws` partition is the commercial partition that most people are used to. There are other partitions though, such as AWS GovCloud (`aws-us-gov`) and AWS China (`aws-cn`)
```

File: docs/_pages/structure.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
+---
+title: "Structure"
+permalink: /structure/
+toc: true
+---
+
+# File Structure
+```
+tardigrade/
+├── aws                                    # AWS parition
```
Comment (lorengordon) [inline docs/_pages/structure.md]:
```suggestion
├── aws                                    # AWS partition
```

File: docs/_pages/structure.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+│   ├── tenant-global.tfvars.yaml          # partition variable values
+│   └── terragrunt.hcl                     # terragrunt configuration
+└── baseline                               # account baseline
+```
+
+## Partition
+In this example, the partition is aws (i.e., `tardigrade/aws`). Generally speaking, most AWS implementations of tardigrade  will only use the `aws` partition. The `aws` parition is the commercial partition that most people are used to. There are other paritions though, such as AWS GovCloud (`aws-us-gov`) and AWS China (`aws-cn`)
+
+### Idempotent Element
+Each directory at this level (e.g., `tardigrade/aws/<accounts>`) represents an individual account. For AWS, the account number is an idempotenet value so we used that to represent individual accounts. The names of these directories do not influence the operation of this framework in any way. They simply serve as a mechanism for developers to easily identify the account.
```
Comment (lorengordon) [inline docs/_pages/structure.md]:
```suggestion
Each directory at this level (e.g., `tardigrade/aws/<accounts>`) represents an individual account. For AWS, the account number is an idempotent value so we used that to represent individual accounts. The names of these directories do not influence the operation of this framework in any way. They simply serve as a mechanism for developers to easily identify the account.
```

File: docs/_pages/structure.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+│   │   ├── base                           # baseline configuration
+│   │   │   ├── tenant-local.auto.tfvars   # defined variables valuables specific to the account
+│   │   │   ├── terragrunt.hcl             # terragrunt configuration for the account
+│   │   └── custom                         # custom configuration
+│   │       ├── terragrunt.hcl             # terragrunt configuration for the account
+│   │       └── variables.tf               # variable declaration specific to the customization
+│   ├── tenant.tf                          # partition provider and variable declarations
+│   ├── tenant-global.tfvars.yaml          # partition variable values
+│   └── terragrunt.hcl                     # terragrunt configuration
+└── baseline                               # account baseline
```
Comment (lorengordon) [inline docs/_pages/structure.md]: this should probably be under "roots/" or "modules/" or similar, as there could be multiple baseline configs handling tf logic.

File: tardigrade/aws/terragrunt.hcl [hunk 6]
```diff
@@ -0,0 +1,12 @@
+remote_state {
+  backend = "s3"
+
+  config = {
+    profile        = "tardigrade"
+    region         = "us-east-1"
+    bucket         = "terragrunt-tardigrade"
```
Comment (lorengordon) [inline tardigrade/aws/terragrunt.hcl]:
```suggestion
    bucket         = "tardigrade-tfstate"
```

File: tardigrade/aws/terragrunt.hcl [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+    key            = "tfstate/${path_relative_to_include()}/terraform.tfstate"
+    encrypt        = true
+    dynamodb_table = "terragrunt-locks-tardigrade"
```
Comment (lorengordon) [inline tardigrade/aws/terragrunt.hcl]:
```suggestion
    dynamodb_table = "tardigrade-tfstate-lock"
```

File: tardigrade/baseline/main.tf [hunk 8]
```diff
@@ -0,0 +1,419 @@
...
+  create_bucket = local.create_keystore_bucket
+  bucket        = var.keystore_bucket
+  region        = var.region
+  policy        = join("", data.template_file.keystore_bucket_policy.*.rendered)
+  versioning    = var.keystore_bucket_versioning
+  tags          = local.tags
+  force_destroy = true
+}
+
+module "keystore" {
```
Comment (lorengordon) [inline tardigrade/baseline/main.tf]:
```suggestion
module "keystore_s3" {
```

**Pull request merged**